File: scripts/executar-usuarios.js
const { createClient } = require('@supabase/supabase-js');

const supabaseUrl = 'https://vzaabtzcilyoknksvhrc.supabase.co';
const serviceRoleKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InZ6YWFidHpjaWx5b2tua3N2aHJjIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc3NDUyNDI0NiwiZXhwIjoyMDkwMTAwMjQ2fQ.b0QCcqqIJMrx8li0g_uRXoJ9z114YWyiHvu5QPjMG7o';

const supabase = createClient(supabaseUrl, serviceRoleKey);

async function adicionarUsuarios() {
  try {
    console.log('🔄 Iniciando adição de usuários...\n');

    // 1️⃣ Criar usuário: Giovanni
    console.log('➕ Criando Giovanni (giovanni.comercialbiasi@gmail.com)...');
    const { data: giovanni, error: errorGiovanni } = await supabase.auth.admin.createUser({
      email: 'giovanni.comercialbiasi@gmail.com',
      password: '1234',
      email_confirm: true,
      user_metadata: {
        nome: 'Giovanni',
        papel: 'user'
      }
    });

    if (errorGiovanni) {
      console.error('❌ Erro ao criar Giovanni:', errorGiovanni.message);
    } else {
      console.log('✅ Giovanni criado com UUID:', giovanni.user.id);
    }

    // 2️⃣ Criar usuário: Jenni
    console.log('\n➕ Criando Jenni (araujojenni2009@gmail.com)...');
    const { data: jenni, error: errorJenni } = await supabase.auth.admin.createUser({
      email: 'araujojenni2009@gmail.com',
      password: '1234',
      email_confirm: true,
      user_metadata: {
        nome: 'Jenni',
        papel: 'user'
      }
    });

    if (errorJenni) {
      console.error('❌ Erro ao criar Jenni:', errorJenni.message);
    } else {
      console.log('✅ Jenni criada com UUID:', jenni.user.id);
    }

    // 3️⃣ Inserir perfis na tabela usuarios
    if (giovanni && !errorGiovanni) {
      console.log('\n📝 Adicionando perfil de Giovanni na tabela usuarios...');
      const { error: errorGiovanniProfile } = await supabase
        .from('usuarios')
        .insert({
          id: giovanni.user.id,
          nome: 'Giovanni',
          email: 'giovanni.comercialbiasi@gmail.com',
          papel: 'user',
          ativo: true
        });

      if (errorGiovanniProfile) {
        console.error('❌ Erro ao adicionar perfil de Giovanni:', errorGiovanniProfile.message);
      } else {
        console.log('✅ Perfil de Giovanni adicionado');
      }
    }

    if (jenni && !errorJenni) {
      console.log('\n📝 Adicionando perfil de Jenni na tabela usuarios...');
      const { error: errorJenniProfile } = await supabase
        .from('usuarios')
        .insert({
          id: jenni.user.id,
          nome: 'Jenni',
          email: 'araujojenni2009@gmail.com',
          papel: 'user',
          ativo: true
        });

      if (errorJenniProfile) {
        console.error('❌ Erro ao adicionar perfil de Jenni:', errorJenniProfile.message);
      } else {
        console.log('✅ Perfil de Jenni adicionado');
      }
    }

    console.log('\n🎉 Operação concluída!');
  } catch (error) {
    console.error('💥 Erro geral:', error);
  }
}

adicionarUsuarios();

File: Comercial/orcamentos/scripts/executar-usuarios.py
import requests
import json

SUPABASE_URL = "https://vzaabtzcilyoknksvhrc.supabase.co"
SERVICE_ROLE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InZ6YWFidHpjaWx5b2tua3N2aHJjIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc3NDUyNDI0NiwiZXhwIjoyMDkwMTAwMjQ2fQ.b0QCcqqIJMrx8li0g_uRXoJ9z114YWyiHvu5QPjMG7o"

headers = {
    "Authorization": f"Bearer {SERVICE_ROLE_KEY}",
    "Content-Type": "application/json",
    "apikey": SERVICE_ROLE_KEY,
}

def criar_usuario(email, password, nome):
    """Criar usuário via Supabase Auth Admin API"""
    url = f"{SUPABASE_URL}/auth/v1/admin/users"
    body = {
        "email": email,
        "password": password,
        "email_confirm": True,
        "user_metadata": {
            "nome": nome,
            "papel": "orcamentista"
        }
    }
    
    response = requests.post(url, json=body, headers=headers)
    
    if response.status_code in [200, 201]:
        user_data = response.json()
        return user_data.get("id"), None
    else:
        return None, response.text

def adicionar_perfil_usuario(user_id, nome, email):
    """Adicionar perfil na tabela usuarios"""
    url = f"{SUPABASE_URL}/rest/v1/usuarios"
    body = {
        "id": user_id,
        "nome": nome,
        "email": email,
        "papel": "orcamentista",
        "ativo": True
    }
    
    response = requests.post(url, json=body, headers=headers)
    
    if response.status_code in [200, 201]:
        return True, None
    else:
        return False, response.text

def main():
    print("🔄 Iniciando adição de usuários...\n")
    
    # Criar Giovanni
    print("➕ Criando Giovanni (giovanni.comercialbiasi@gmail.com)...")
    giovanni_id, erro_giovanni = criar_usuario(
        "giovanni.comercialbiasi@gmail.com", 
        "1234", 
        "Giovanni"
    )
    
    if erro_giovanni:
        print(f"❌ Erro ao criar Giovanni: {erro_giovanni}")
    else:
        print(f"✅ Giovanni criado com UUID: {giovanni_id}")
        
        # Adicionar perfil
        print("📝 Adicionando perfil de Giovanni na tabela usuarios...")
        sucesso, erro = adicionar_perfil_usuario(giovanni_id, "Giovanni", "giovanni.comercialbiasi@gmail.com")
        if sucesso:
            print("✅ Perfil de Giovanni adicionado")
        else:
            print(f"❌ Erro ao adicionar perfil: {erro}")
    
    # Criar Jenni
    print("\n➕ Criando Jenni (araujojenni2009@gmail.com)...")
    jenni_id, erro_jenni = criar_usuario(
        "araujojenni2009@gmail.com", 
        "1234", 
        "Jenni"
    )
    
    if erro_jenni:
        print(f"❌ Erro ao criar Jenni: {erro_jenni}")
    else:
        print(f"✅ Jenni criada com UUID: {jenni_id}")
        
        # Adicionar perfil
        print("📝 Adicionando perfil de Jenni na tabela usuarios...")
        sucesso, erro = adicionar_perfil_usuario(jenni_id, "Jenni", "araujojenni2009@gmail.com")
        if sucesso:
            print("✅ Perfil de Jenni adicionado")
        else:
            print(f"❌ Erro ao adicionar perfil: {erro}")
    
    print("\n🎉 Operação concluída!")

if __name__ == "__main__":
    main()
